driver/src/driver/stablex_driver.rs

```diff
@@ -271,4 +310,49 @@ mod tests {
                 .was_called_with((batch - 1, Any, Any, Any))
         ));
     }
+
+    #[test]
+    fn test_does_not_submit_solution_for_which_validation_failed() {
```

Comment:
A sneaky unrelated test case :P I'm totally cool with including this here though since it is vaguely related.